chainer/links/model/vision/resnet.py

```diff
@@ -0,0 +1,447 @@
+        x = Variable(self.xp.asarray(x))
+        return self(x, layers=layers, test=test)
+
+    def predict(self, images, oversample=True):
+        """Computes all the probabilities of given images.
+
+        Args:
+            image (iterable of PIL.Image or numpy.ndarray): Input images.
+            oversample (bool): If ``True``, it avarages results across
+                center, corners, and mirrors. Otherwise, it uses only the
+                center.
+
+        Returns:
+            ~chainer.Variable: Output that contains the class probabilities
+            of given images.
+
+        """
+
+        x = concat_examples([prepare(img, size=(256, 256)) for img in images])
+        if oversample:
+            x = imgproc.oversample(x, crop_dims=(224, 224))
+        else:
+            x = x[:, :, 16:240, 16:240]
+        x = Variable(self.xp.asarray(x))
+        y = self(x, layers=['prob'])['prob']
+        if oversample:
+            n = y.data.shape[0] // 10
+            y_shape = y.data.shape[1:]
+            y = reshape(y, (n, 10) + y_shape)
+            y = sum(y, axis=1) / 10
+        return y
+
+
+def prepare(image, size=(224, 224)):
+    """Converts the given image to the numpy array for ResNets.
+
+    Note that you have to call this method before ``__call__``
+    because the pre-trained resnet model requires to resize the given
+    image, covert the RGB to the BGR, subtract the mean,
+    and permute the dimensions before calling.
+
+    Args:
+        image (PIL.Image or numpy.ndarray): Input image.
+            If an input is ``numpy.ndarray``, its shape must be
+            ``(height, width)``, ``(height, width, channels)``,
+            or ``(channels, height, width)``.
+        size (pair of ints): Size of converted images.
+            If ``None``, the given image is not resized.
+
+    Returns:
+        numpy.ndarray: The converted output array.
+
+    """
+
+    if not available:
+        raise ImportError('PIL cannot be loaded. Install Pillow!\n'
+                          'The actual import error is as follows:\n' +
+                          str(_import_error))
+    if isinstance(image, numpy.ndarray):
+        if image.ndim == 3:
+            if image.shape[0] == 1:
+                image = image[0, :, :]
+            elif image.shape[0] == 3:
+                image = image.transpose((1, 2, 0))
+        image = Image.fromarray(image.astype(numpy.uint8))
+    image = image.convert('RGB')
+    if size is not None:
+        image = image.resize(size)
+    image = numpy.asarray(image, dtype=numpy.float32)
+    image = image[:, :, ::-1]
+    # NOTE: in the original paper they subtract a fixed mean image,
+    #       however, in order to support arbitrary size we instead use the
+    #       mean pixel (rather than mean image) as with VGG team. The mean
+    #       value used in ResNet is slightly different from that of VGG16.
+    image -= numpy.array(
+        [103.063,  115.903,  123.152], dtype=numpy.float32)
```

Comment:
> For greyscale images converted to RGB, converted to BGR, the image statistics is different.

Really? In this case each channel in the gray image must be the same value, so I think it is ok to subtract the above mean.